File: app/src/main/java/net/bible/service/db/bookmark/LabelDto.java
/*
 * Copyright (c) 2018 Martin Denham, Tuomas Airaksinen and the And Bible contributors.
 *
 * This file is part of And Bible (http://github.com/AndBible/and-bible).
 *
 * And Bible is free software: you can redistribute it and/or modify it under the
 * terms of the GNU General Public License as published by the Free Software Foundation,
 * either version 3 of the License, or (at your option) any later version.
 *
 * And Bible is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along with And Bible.
 * If not, see http://www.gnu.org/licenses/.
 *
 */

package net.bible.service.db.bookmark;


import androidx.annotation.NonNull;

import net.bible.android.BibleApplication;
import net.bible.android.activity.R;
import net.bible.android.control.bookmark.BookmarkStyle;

import org.apache.commons.lang3.StringUtils;

/**
 * @author Martin Denham [mjdenham at gmail dot com]
 */
public class LabelDto implements Comparable<LabelDto> {
	private Long id;
	private String name;
	private BookmarkStyle bookmarkStyle;

	public LabelDto() {
	}

	public LabelDto(Long id, String name, BookmarkStyle bookmarkStyle) {
		this.id = id;
		this.name = name;
		this.bookmarkStyle = bookmarkStyle;
	}

	@Override
	public String toString() {
		return getName();
	}

	/* (non-Javadoc)
	 * @see java.lang.Object#hashCode()
	 */
	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((id == null) ? 0 : id.hashCode());
		return result;
	}
	
	/* (non-Javadoc)
	 * @see java.lang.Object#equals(java.lang.Object)
	 */
	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		LabelDto other = (LabelDto) obj;
		if (id == null) {
			if (other.id != null)
				return false;
		} else if (!id.equals(other.id))
			return false;
		return true;
	}

	public Long getId() {
		return id;
	}
	public void setId(Long id) {
		this.id = id;
	}
	public String getName() {
		if(bookmarkStyle == BookmarkStyle.SPEAK) {
			return BibleApplication.getApplication().getString(R.string.speak);
		}
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}

	public BookmarkStyle getBookmarkStyle() {
		return bookmarkStyle;
	}

	public void setBookmarkStyle(BookmarkStyle bookmarkStyle) {
		this.bookmarkStyle = bookmarkStyle;
	}

	public String getBookmarkStyleAsString() {
		if (bookmarkStyle==null) {
			return null;
		} else {
			return bookmarkStyle.name();
		}
	}

	public void setBookmarkStyleFromString(String bookmarkStyle) {
		if (StringUtils.isEmpty(bookmarkStyle)) {
			this.bookmarkStyle = null;
		} else {
			this.bookmarkStyle = BookmarkStyle.valueOf(bookmarkStyle);
		}
	}

	@Override
	public int compareTo(@NonNull LabelDto another) {
		return name.compareToIgnoreCase(another.name);
	}
}

File: app/src/main/java/net/bible/service/db/bookmark/LabelDto.kt
/*
 * Copyright (c) 2020 Martin Denham, Tuomas Airaksinen and the And Bible contributors.
 *
 * This file is part of And Bible (http://github.com/AndBible/and-bible).
 *
 * And Bible is free software: you can redistribute it and/or modify it under the
 * terms of the GNU General Public License as published by the Free Software Foundation,
 * either version 3 of the License, or (at your option) any later version.
 *
 * And Bible is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along with And Bible.
 * If not, see http://www.gnu.org/licenses/.
 *
 */
package net.bible.service.db.bookmark

import net.bible.android.BibleApplication.Companion.application
import net.bible.android.activity.R
import net.bible.android.control.bookmark.BookmarkStyle
import org.apache.commons.lang3.StringUtils

/**
 * @author Martin Denham [mjdenham at gmail dot com]
 */
class LabelDto : Comparable<LabelDto> {
    var id: Long? = null
    var name: String? = null
		get() = if (bookmarkStyle == BookmarkStyle.SPEAK) {
			application.getString(R.string.speak)
		} else field

    var bookmarkStyle: BookmarkStyle? = null

    constructor()
	constructor(id: Long?, name: String?, bookmarkStyle: BookmarkStyle?) {
        this.id = id
        this.name = name
        this.bookmarkStyle = bookmarkStyle
    }

    override fun toString(): String {
        return name!!
    }

    /* (non-Javadoc)
	 * @see java.lang.Object#hashCode()
	 */
    override fun hashCode(): Int {
        val prime = 31
        var result = 1
        result = prime * result + if (id == null) 0 else id.hashCode()
        return result
    }

    /* (non-Javadoc)
	 * @see java.lang.Object#equals(java.lang.Object)
	 */
    override fun equals(obj: Any?): Boolean {
        if (this === obj) return true
        if (obj == null) return false
        if (javaClass != obj.javaClass) return false
        val other = obj as LabelDto
        if (id == null) {
            if (other.id != null) return false
        } else if (id != other.id) return false
        return true
    }

    val bookmarkStyleAsString: String?
        get() = if (bookmarkStyle == null) {
            null
        } else {
            bookmarkStyle!!.name
        }

    fun setBookmarkStyleFromString(bookmarkStyle: String?) {
        if (StringUtils.isEmpty(bookmarkStyle)) {
            this.bookmarkStyle = null
        } else {
            this.bookmarkStyle = BookmarkStyle.valueOf(bookmarkStyle!!)
        }
    }

    override fun compareTo(another: LabelDto): Int {
        return name!!.compareTo(another.name!!, ignoreCase = true)
    }
}
